k2/csrc/fsa.h

```diff
@@ -11,6 +11,8 @@
 #include <utility>
 #include <vector>
 
+#include "k2/csrc/util.h"
```

Comment:
Is this header needed here?

Reply:
Maybe we can move the hash function to `fsa.h`.

Reply:
No,  I think it's better to put this in a new file, as `hash_combine` is not just for `ArcHash` and may be called by other struct, such as `std::pair`.

Note that there is a hash function named `ArcHash` in fsa.h
https://github.com/danpovey/k2/blob/1e362c1b3b603c57cde6bbbb08153f34ba99a88c/k2/csrc/fsa.h#L49-L55